pkg/utils/utils.go

```diff
@@ -0,0 +1,58 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/containers/image/pkg/sysregistries"
+)
+
+// CheckConfigFiles log messages if one or more files to set up buildah do not exist
+func CheckConfigFiles() error {
+	errMsg := ""
+	registriesConfPath := sysregistries.RegistriesConfPath(nil)
+	_, err := os.Stat(registriesConfPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			errMsg = fmt.Sprintf("Required file %s not found.\n", registriesConfPath)
+		} else {
+			errMsg = fmt.Sprintf("error checking %s, %s", registriesConfPath, err)
```

Comment:
The code in **EDIT<s>`registries.conf`</s>`c/image/pkg/sysregistriesv2` explicitly supports the situation when the file is missing. It’s not at all obvious to me that buildah should enforce its existence independently — if it is _that_ necessary, `c/image/pkg/sysregistriesv2` should fail if it is missing as well.

To the extent this is about https://github.com/containers/buildah/issues/1307 , buildah already has the `searchRegistriesWereUsedButEmpty` boolean in the search path, so isn’t this just a matter of propagating/displaying that error correctly, or maybe using the same boolean on other code paths (or consolidating them)?

Reply:
@TomSweeneyRedHat 

The question is how user friendly do we want to make the tool. Can the tools diagnose the situation, where the user is likely to get a bad experience.  People are packaging up buildah either from source or for other distributions and are suprised when they don't work correctly.  Then they have to find random config files to make them run correctly.

So the goal was to make this up front to the user.

Lets examine the errors.
# No registries.conf
```
 buildah from fedora
(image name "fedora" is a short name and no search registries are defined in /etc/containers/registries.conf): while pulling "fedora" as "localhost/fedora": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
```

# No Policy.json
```
buildah from fedora:f28
The following failures happened while trying to pull image specified by "fedora:f28" based on search registries in /etc/containers/registries.conf:
* "localhost/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
* "docker.io/library/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
* "registry.fedoraproject.org/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
* "quay.io/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
* "registry.access.redhat.com/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
* "registry.centos.org/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
```

# No /etc/containers/storage.conf

```
# buildah from fedora:28
Getting image source signatures
Copying blob 012cb44e3253: 85.49 MiB / 85.49 MiB [=========================] 26s
Copying config 4c5b21baa7a0: 1.99 KiB / 1.99 KiB [==========================] 0s
Writing manifest to image destination
Storing signatures
fedora-working-container-1
```

Reply:
> # No Policy.json
> ```
> buildah from fedora:f28
> The following failures happened while trying to pull image specified by "fedora:f28" based on search registries in /etc/containers/registries.conf:
> * "localhost/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
> * "docker.io/library/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
> * "registry.fedoraproject.org/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
> * "quay.io/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
> * "registry.access.redhat.com/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
> * "registry.centos.org/fedora:f28": error obtaining default signature policy: open /etc/containers/policy.json: no such file or directory
> ```

BTW this could be easily improved (and made a bit more efficient, avoiding repeated re-parsings of the file, and in the future perhaps redundant up GPG home directory creations) by moving `signature.NewPolicyContext` from `pullImage` out of the loop to `resolveImage`